src/rtcp/receiverReport.ts

```diff
@@ -0,0 +1,364 @@
+	}
+
+	/*
+	 * Get delay since last Sender Report.
+	 */
+	getDelaySinceLastSR(): number
+	{
+		return this.buffer.readUIntBE(20, 4);
+	}
+
+	/*
+	 * Set delay since last Sender Report.
+	 */
+	setDelaySinceLastSR(dlsr: number): void
+	{
+		this.buffer.writeUIntBE(dlsr, 20, 4);
+	}
+}
+
+export class ReceiverReportPacket extends RtcpPacket
+{
+	// Receiver Reports.
+	private reports: ReceiverReport[] = [];
+	// Whether serialization is needed due to modifications.
+	private serializationNeeded: boolean = false;
+
+	// Packet Type.
+	static packetType = PacketType.RR;
+
+	/**
+	 * @param buffer - If given it will be parsed. Otherwise an empty RTP packet
+	 *   will be created.
+	 */
+	constructor(buffer?: Buffer)
+	{
+		super(ReceiverReportPacket.packetType);
+
+		// If no buffer is given, create an empty one with minimum required length.
+		if (!buffer)
+		{
+			this.buffer = Buffer.alloc(FIXED_HEADER_LENGTH);
+			this.writeCommonHeader();
+
+			return;
+		}
+
+		if (!isRtcp(buffer))
+		{
+			throw new TypeError('invalid RTCP packet');
+		}
+
+		let count = RtcpPacket.getCount(buffer);
+
+		if (buffer.length < FIXED_HEADER_LENGTH + (count * REPORT_LENGTH))
+		{
+			throw new TypeError('buffer is too small');
+		}
+
+		while (count-- > 0)
+		{
+			const report = new ReceiverReport(
+				buffer.slice(FIXED_HEADER_LENGTH + (this.reports.length * REPORT_LENGTH))
+			);
+
+			this.addReport(report);
+		}
+
+		// Store a buffer within the packet boundaries.
+		this.buffer = buffer.slice(
+			undefined, FIXED_HEADER_LENGTH + (this.reports.length * REPORT_LENGTH));
+	}
+
+	/**
+	 * @ignore
+	 */
+	dump(): any
+	{
+		return {
+			... super.dump(),
+			ssrc    : this.getSsrc(),
+			reports : this.reports.map((report) => report.dump())
+		};
+	}
+
+	/**
+	 * Get the internal buffer containing the serialized RTP binary packet.
+	 */
+	getBuffer(): Buffer
+	{
+		if (this.serializationNeeded)
+		{
+			this.serialize();
+		}
+
+		return this.buffer;
+	}
+
+	/*
+	 * Get sender SSRC.
+	 */
+	getSsrc(): number
+	{
+		return this.buffer.readUInt32BE(4);
+	}
+
+	/*
+	 * Set sender SSRC.
+	 */
+	setSsrc(ssrc: number)
+	{
+		this.buffer.writeUInt32BE(ssrc, 4);
+	}
+
+	/*
+	 * Get Receiver Reports.
+	 */
+	getReports(): ReceiverReport[]
+	{
+		return this.reports;
+	}
+
+	/*
+	 * Add a Receiver Report.
+	 */
+	addReport(report: ReceiverReport): void
+	{
+		this.reports.push(report);
+		this.serializationNeeded = true;
+	}
+
+	/**
+	 * Apply pending changes into the packet and serialize it into a new internal
+	 * buffer (the one that [[getBuffer]] will later return).
+	 *
+	 * **NOTE:** In most cases there is no need to use this method. It must be
+	 * called only if the application retrieves information from the packet (by
+	 * calling [[getBuffer]], [[getReports]], etc) and modifies the obtained
+	 * buffers in place. However, it's recommended to use the existing setter
+	 * methods instead ([[addReport]], etc).
+	 *
+	 */
+
+	public serialize(): void
+	{
```

Comment:
Remove blank line and unneeded `public` keyword:

```suggestion
	 */
	serialize(): void
	{
```